lib/mergeViewsInCollections.js

```diff
@@ -17,26 +17,19 @@
 'use strict'
 
 module.exports = function mergeViewsInCollections(collections, views) {
-  const collectionsObject = collections.reduce((acc, collection) => {
-    const coll = (acc[collection.name] || []).concat(collection)
-    return {
-      ...acc,
-      [collection.name]: coll,
-    }
-  }, {})
-
-  for (const view of views) {
-    const collectionConfig = collectionsObject[view.name]
-    if (collectionConfig) {
-      collectionsObject[view.name] = collectionConfig.map(coll => ({
-        ...coll,
-        source: view.source,
+  return views.reduce((acc, view) => {
+    const { name: viewName } = view
+    const associatedCollectionIndex = collections.findIndex(({ name: collectionName }) => collectionName === viewName)
+    if (associatedCollectionIndex >= 0) {
+      acc[associatedCollectionIndex] = {
+        ...collections[associatedCollectionIndex],
+        ...view,
         // Mandatory to initialize the collection in Mongo as a MongoDB View
         type: 'view',
-        pipeline: view.pipeline,
-      }))
+      }
     }
-  }
-
-  return Object.values(collectionsObject).flat()
+    return acc
+  }, [
+    ...collections,
+  ])
```

Comment:
It looks confusing to me the idea of using a `reduce` loop on the list of views to update the collections, which is a different list. 

I'd use a forEach and create a shallow copy of `collections` before. Something like this:
```
function mergeViewsInCollections(collections, views) {
  const mergedCollections = [...collections];
  
  views.forEach(view => {
    const { name: viewName } = view;
    const associatedCollectionIndex = collections.findIndex(({ name: collectionName }) => collectionName === viewName);
    
    if (associatedCollectionIndex >= 0) {
      mergedCollections[associatedCollectionIndex] = {
        ...collections[associatedCollectionIndex],
        ...view,
        type: 'view',
      };
    }
  });
  
  return mergedCollections;
}
```

Reply:
At the end here I decided to use a for loop, just to be faster. Do you agree with my decision?